### DISCUSSION TITLE
Web Settings problems and improvements

### DISCUSSION THREAD

**Discussion by Mik-S-UMS** [UniversalMediaServer/UniversalMediaServer#3773] [Ideas]
I found some problems when helping a user on the forum so spent a few hours going though all the settings to find any problems. 
This is what I have found ranging from nit-picky to things not working correctly, plus a few suggestions. 

Maybe others can add their suggestions too.

I thought it might be better to put here instead of making a load of issues. I will make an issue about some settings not working though.

Version used 13.1.0-SNAPSHOT UMS-3130.exe, some may have already been fixed in main release.

### Server settings:
Bottom of letters like "g" missing in section heads
	
![Missing bottom of g](https://user-images.githubusercontent.com/66766335/212196723-1f1fdf0a-83a2-499f-aed4-5fd687eed257.jpg)

Would be better to have the save button greyed out if there has not been any change so you can see if you need to save or not. 
(same for shared content)

### General Settings
**Services**
_Web player_
`Allow media download` does not do anything, download button still present and functional when set to false

### Navigation Settings
**General Settings**
`Enable the cache` is linked with `Show the "Media Library" folder`
`Fully played action` shows blank when set
If set to move, the Destination folder is not shown unless you re-select move in action
This is what it looks like when opening settings with that option set
![image](https://user-images.githubusercontent.com/66766335/212200978-8dfbb0b7-faa8-4c55-856e-d006ff872349.png)

And this is what it should look like
![image](https://user-images.githubusercontent.com/66766335/212200999-7410a08d-8d42-41f3-b541-b0d1d38a7c3e.png)

**File sorting / Naming**
Would be good to have an option to prepend a number to each file so renderers (like PS4 and Samsung TVs) that will sort them alphabetically, ignoring the order sent by UMS, will show in the chosen order. 
This would help tidy up the `#--Transcode--#` folder too, having `00 - No Transcoding`  will always make it the first option.

**Virtual Folders / Files**
`Show the "Media Library" folder` changes `use_cache` in conf 
(I think the Media Library folder should be shown in the web player even if disabled here)
`Show the "Live Subtitles" folder` does not work
		
### Transcoding settings
**Common transcoding settings**
**video settings**
`Transcoding quality (MPEG-2):` is a text box and can be set to anything, Was a drop-down in old GUI
`Transcoding quality (H.264):` Same as above
		
**Audio settings**
`Audio language priority:` changes nothing in settings, commented out in conf but still does not work if uncommented.

**Subtitles settings**
`Subtitles language priority:` does nothing, commented out in conf
`Forced language:` commented out in conf but does add a new `forced_subtitle_language` if set 		
`Audio/subtitles language priority:` commented out in conf but adds the setting at the end. Does not show default setting. 
This should now be improved with a more user friendly way of selecting language pairs from a list instead of using codes
`Alternate subtitles folder:` can't delete if set, unable to type in box
`Specify TrueType font:` can't pick a font, only select a directory. This should bring a list of system fonts. Can't choose a windows font either in the old GUI. Also can't delete this once a directory is set, Default font should be shown too.
`Font scale` could do with arrows to alter like with "Font outline" etc
It would be good if there was a preview to see what these settings will look like in the video.

**Transcoding Engines**
_MEncoder video_
`Remux videos with tsMuxeR when possible instead of transcoding `does nothing

_VLC Video_
`vlc_use_experimental_codecs` and `vlc_audio_sync_enabled` commented out in conf but new ones added if set.

_VLC Web Video_
Uses the same settings as VLC Video so seems redundant

When disabling or enabling multiple engines the tool-tips can stay and won't go away, 
![Sticky tool tips on renderer activation](https://user-images.githubusercontent.com/66766335/212202775-e7e0df84-82eb-44a9-96c0-e94024b4087e.jpg)

even after collapsing and expanding the section
![Sticky tool tips on renderer activation2](https://user-images.githubusercontent.com/66766335/212202805-9f8a342d-076c-4c2a-8dc3-2e27f5a72d1b.jpg)

### Needs documentation / tool-tips
Not sure what these do exactly or how to use them, needs more documentation
**General settings** > **Services** > **Web player** - `Use authentication service` and `Can control other devices`
**Navigation settings** > **General settings** - `Use target file for symbolic links`
Many options in transcoding settings have no tool tips but are documented in UMS.conf

### Selected directory: 
Browsing folders when selecting one in the settings is very slow to get the folder list.
Default location for selecting directory should be computer listing all drives, with shortcuts to common places like Videos, Music etc
Should be possible to type and edit a path too.

### Tools:
`Restart Server` `Restart Application` and `Shutdown Application`
The confirmation window has a scroll bar and another appears when clicking the buttons.
![scroll bar on restart UMS box](https://user-images.githubusercontent.com/66766335/212197516-4e161a07-2c8b-48d0-8624-631c0d05bd66.jpg)
![scroll bars on restart UMS box when clicking](https://user-images.githubusercontent.com/66766335/212197689-09681c6f-9789-4a3f-9fa4-fe622e9e90b5.jpg)

### Logs:
Don't see a way to put UMS in trace mode via the browser. Can see the filter option in settings but unable to select `TRACE` or `ALL`. 
This also does not change back to `INFO` if that is selected when in `TRACE` mode. 
If this is not the way to change logging mode but is a filter then that seems redundant as there is already a filter.

Logs only show 1 page of the first 300 lines but you have to click on it first to show then it sometimes disappears, this should be automatically shown when the logs are opened, and better if the end of the log is shown instead of the start.
![only 1 page of logs](https://user-images.githubusercontent.com/66766335/212196877-d45d9988-340c-448c-beec-f4e7c5364e65.jpg)

The filter options would be easier to use if not in its own window similar to how the old GUI has it. Some of the GUI elements seem a little squashed together too.
![image](https://user-images.githubusercontent.com/66766335/212197077-535a3bb9-fac7-4e8d-afc3-becadb121a78.png)

Making trace logs should be made easier instead of buried away in the **Tools** > **View logs** > **Options** > `create trace logs`, too many clicks to get there.

One of the options in Tools should be `Create trace logs for reporting issues`
There should also be some sort of indicator (message in header or different colour to the GUI) that UMS is running in temporary trace mode and an easy to access `Pack debug files` button on the main screen. 
Once debug files have been packed then UMS should return to its previous logging mode.

There is no need for `Pack debug files` button when not in trace mode as those logs will not be very helpful when looking for help on the forum.

The error notification that links to GitHub should also upload the logs and auto-fill the Issue with some information if that is possible. I know there has been several blank issues created with the `Launch GUI in browser on startup` bug

### Shared Content:
Would be better to treat all folders as virtual folders to allow the ability to give them names and have more flexibility.

Unable to set contents of virtual folders as `monitored`, `mark as fully played` or `unplayed` despite having those options in the shared.conf
Would make more sense to move those options in SHARED.conf and have the child section inherit them. e.g. something like this
```
 {
    "name": "TV Episodes",
    "monitored": true,
    "metadata": true,
    "addToMediaLibrary": true,
    "active": true,
    "type": "VirtualFolder"
    "childs": [
      {
        "file": "K:\\Users\\Mik\\Videos\\Episodes",
        "file": "L:\\More eps"
      }
    ],
  },
```
 
Instead of 
```
  {
    "name": "TV Episodes",
    "childs": [
      {
        "file": "K:\\Users\\Mik\\Videos\\Episodes",
        "monitored": true,
        "metadata": true,
        "active": true,
        "type": "Folder"
      },
      {
        "file": "L:\\More eps",
        "monitored": true,
        "metadata": true,
        "active": true,
        "type": "Folder"
      }
    ],
    "addToMediaLibrary": true,
    "active": true,
    "type": "VirtualFolder"
  },
```
  
The shared content is a perfect place to add in security settings in a user friendly way that have been lacking, broken, and hard to setup as were not available in the GUI. 
There should also be an option not to look for meta-data for certain folders for privacy, things like private videos or adult content. Even logging of these folders should be turned off.

Individual options for Folders could be added too. 
For example fully played actions like move you might not want a file to be moved across to other drives but to a watched folder on the same drive. Another folder you might not want it move at all.

Virtual folders could be set up for Anime, one folder to have English audio and another folder to play the same videos with Japanese Audio and English subs, depending on if you want the SUB or DUB then you don't need to hunt for the right option in the `#--Transcode--#` folder amongst all the other possibilities.

### Web Player:
**Videos**
Videos do not auto play when you click on them, you have to click on them again once the player comes up, I don't know if this is a Firefox thing as I have disabled auto-play for most sites but cannot set an exception for UMS as it has not requested that permission. But still does not auto-play if I allow it browser wide.

**Series**
In the overview for a series the episodes could be divided for each season/series instead of grouping them together
![image](https://user-images.githubusercontent.com/66766335/212209158-c3150bbb-4418-4ffc-9608-b28841a64daf.png)
On second row the last 3 are series 2, would look nicer if they were separated with a header saying **Season 2**
Also There is no need to show the full filename here, would be better to name them `Episode # - Episode name` here.

It also takes too many clicks to play from the overview page. From that screenshot I click on episode 1 which brings up
![image](https://user-images.githubusercontent.com/66766335/212210408-9626cafd-6e4d-43a8-b66a-ce1689ba7844.png)

Clicking play brings up the player
![image](https://user-images.githubusercontent.com/66766335/212210495-00f87861-d83c-4746-a0f5-709fb7a97a16.png)
Then you have to click again to play.

That example may not be the best example as with another series clicking on an episode brings up more information.
![image](https://user-images.githubusercontent.com/66766335/212211291-c22fcdd8-e738-4b66-b68a-889d36e57bba.png)

But maybe it would be better to use some of the unused space on the left side to display this alongside playing that episode straight away

You could add a tool-tip with the synopsis of an episode when you hover over it.

An **Auto-play Next Episode** option when played from this series overview page would be good for binge watching a whole series needing no interaction.

**Audio**
I think some work need to be done on playing audio. 
At the moment it is using the video player which is not a great way to listen music on, not helped by lack of auto-play.
It would be better if it had some sort of interface like iTunes, Spotify or other smiler music player with where you can see all albums by an artist, all tracks in an album, high quality album art etc. 
Maybe even a mini-player at the top of the directory listing to let you continue to play the album/play-list when browsing other folders until you start playing something else.

The way the music is organized can be refined too. If I select an Artist then all their music is jumbled together in the main window. 
![image](https://user-images.githubusercontent.com/66766335/212215287-c65bc645-7a98-4397-8837-56d4ae4e1139.png)
This should be listed by album, sorted by release date, with just one picture for the album art and the files sorted by disk number then track number, just shown as text with a play button beside each one in the web interface.
Of course the sorting should be customisable too instead of the fixed options.

These mp3 files are being transcoded too which seems unnecessary. I would have thought that would not be needed for mp3s.

**Comment by SubJunk** [comment]
This is really great investigation and documenting work, thanks
